src/interfaces.ts

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+
+import Tester from './tester';
+import ConfigurationClass from './ConfigurationClass';
+
+export type ComponentClass = React.FC | { new(): Component<any> };
+
+export interface IMountOps {
+  async?: boolean;
+}
+
+export interface IWrapper {
+  component: ComponentClass;
+  name: string;
+  props: object;
+  renderChildren?: boolean;
+}
+
+export interface IHook extends IWrapper {
+  [key: string]: any;
+  onBeforeMount: (tester: Tester, ops: IMountOps) => void;
```

Comment:
In the code I use:
`await hook.onBeforeMount(tester, mountOpts);`
I'm fine with keeping it short, but please fix the typo ;) `ops` -> `opts`